RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -122,7 +122,7 @@ extension PurchaseHandler {
             return false
         }
 
-        self.track(.close(data.withCurrentDate()))
+        self.track(.close(data.withNewID().withCurrentDate()))
```

Comment:
Need this otherwise the `id` was the same for each event in the session 😅

Reply:
This is a smell in how we're using the type.

I think to prevent this we should put the `Event.ID` inside of `PaywallEvent` not `PaywallEvent.Data`. What do you think?